javascript/javascript1/week4/voice-assistant.js

```diff
@@ -0,0 +1,219 @@
+                if(nameArray.includes(nameString.toLowerCase()) )
+                {
+                    console.log('Name already exists, so skipping it');
+                }
+                else
+                {
+                    console.log('name do not exists, so adding to the list');
+                    nameArray.push(nameString.toLowerCase()); // add to the array
+                    console.log(`Nice to meet you:  ${nameString} !`);
+                    //console.log(`Total Names in the array ${nameArray.length}`);  
+                }
+        
+            }
+            break;
+
+        //--------------------------------------------------------------
+
+            case 'Add':    
+                
+                if(wordString.length > 4)
+                {
+                    nameString2 = wordString;
+                    nameString2.splice(0, 1); // remove first element
+                    nameString2.splice(-1, 1); // remove last element
+                    nameString2.splice(-1, 1); // remove 2nd last element, whih is infact last now
+                    nameString2 = nameString2.join(' '); // combine element to form the string
+                }
+                else
+                {
+                    nameString2 = wordString[1];
+                }
+
+                todoArray.push(nameString2.toLowerCase());
+                console.log(`Added ${nameString2} to myToDo`);
+                //console.log(`Total items in the ToDo  ${todoArray.length}`);
+                break;
+
+            case 'Remove':    
+                if(wordString.length > 4)
+                {
+                    nameString2 = wordString;
+                    nameString2.splice(0, 1); // remove first element
+                    nameString2.splice(-1, 1); // remove last element
+                    nameString2.splice(-1, 1); // remove 2nd last element, whih is infact the last now
+                    nameString2 = nameString2.join(' '); // combine element to form the string
+                }
+                else
+                {
+                    nameString2 = wordString[1];
+                }
+
+                 // check if the item is in the list, before removing it.If exists then remove it.
+                if(todoArray.includes(nameString2.toLowerCase()))
+                {
+                    itemIndex = todoArray.indexOf(nameString2);
+                    todoArray.splice(itemIndex, 1) ; 
+                    console.log(`${nameString2} is removed from myToDo`);              
+                }
+                else 
+                {    
+                    // if not in the list, print invalid item to be removed
+                    // if item exist in the list, remove it :-) 
+                    console.log(`Invalid item to be removed`);
+                }
+                break;
+
+            
+            //----------------------------------
+            case 'What':    
+                nameString2 = wordString;
```

Comment:
Still shouldn't be `nameString2`. 
You could just assign a new variable right here: 
`let nameCommand = wordString`